regex-base.cabal

```diff
@@ -83,4 +94,4 @@ library
   if !impl(ghc >= 8)
       build-depends: fail == 4.9.*
 
-  ghc-options: -Wall -O2
+  ghc-options: -Wall
```

Comment:
This should be mentioned in changelog

Reply:
What's the motivation behind this change? Was its impact tested/benchmarked?

Reply:
@phadej : Added to changelog.

@Bodigrim: The motivation is that `-O2` is generally discouraged in cabal files.  `cabal` gives a warning if you have it.  This package does not contain any heavy code really, just a bit of administration.  See also https://github.com/haskell-hvr/regex-tdfa/issues/7.

Reply:
Afaiu, those flags are tossed out in compilation anyway.